File: Iniciante/URI 1046.r
input = file('stdin', 'r')

line = strsplit(readLines(input, n=1), " ")
hInicial = as.integer(line[[1]][1])
hFinal = as.integer(line[[1]][2])

if (hInicial > hFinal) { 
    write(sprintf("O JOGO DUROU %d HORA(S)", 24 - (hInicial - hFinal)), '')
} else if (hFinal > hInicial) {
    write(sprintf("O JOGO DUROU %d HORA(S)", (hFinal - hInicial)), '')
} else {
   write(sprintf("O JOGO DUROU 24 HORA(S)"), '');
}

File: Iniciante/URI 1046.php
<?php
    $line = explode(" ", trim(fgets(STDIN)));
    $hInicial = intval($line[0]);
    $hFinal = intval($line[1]);
    
    if ($hInicial > $hFinal) { 
        echo "O JOGO DUROU " . (24 - ($hInicial - $hFinal)) . " HORA(S)\n";
    } else if ($hFinal > $hInicial) {
        echo "O JOGO DUROU " . ($hFinal - $hInicial) . " HORA(S)\n";
    } else {
       echo "O JOGO DUROU 24 HORA(S)\n";
    }
?>